cdap-data-fabric/src/main/java/co/cask/cdap/store/OwnerStore.java

```diff
@@ -0,0 +1,204 @@
+ * associated owner's principal. Note: An absence of an entry in this table for an {@link EntityId} does not
+ * signifies that the entity does not exists. The owner information is only stored if an owner was provided during
+ * creation time else the owner information is non-existent which signifies that the entity own is default CDAP owner.
+ * </p>
+ * <p>
+ * <p>
+ * The owner's principal must be in the of the following Kerberos name types:
+ * <ul>
+ * <li>KRB_NT_PRINCIPAL:  Just the name of the principal as in DCE, or for users. For example: alice@REALM</li>
+ * <li>KRB_NT_SRV_HST:  Service with host name as instance(telnet, rcommands). For example alice/hostname@REALM</li>
+ * </ul>
+ * Refer to <a href=https://tools.ietf.org/html/rfc4120#section-7.5.8>Name Types</a> documentation for details on
+ * Kerberos Name Types.
+ * </p>
+ */
+public class OwnerStore {
+  private static final String OWNER_PREFIX = "o";
+  // currently, we only leverage one column of the table. However, not using KeyValueTable, so that being able to use
+  // additional columns in the future is simple
+  private static final byte[] COL = Bytes.toBytes("c");
+  private static final DatasetId DATASET_ID = NamespaceId.SYSTEM.dataset("owner.meta");
+  private static final DatasetProperties DATASET_PROPERTIES =
+    DatasetProperties.builder().add(Table.PROPERTY_CONFLICT_LEVEL, ConflictDetection.COLUMN.name()).build();
+
+  private final DatasetFramework datasetFramework;
+  private final Transactional transactional;
+
+  @Inject
+  OwnerStore(DatasetFramework datasetFramework, TransactionSystemClient txClient) {
+    this.datasetFramework = datasetFramework;
+    this.transactional = Transactions.createTransactionalWithRetry(
+      Transactions.createTransactional(new MultiThreadDatasetCache(new SystemDatasetInstantiator(datasetFramework),
+                                                                   txClient, DATASET_ID.getParent(),
+                                                                   Collections.<String, String>emptyMap(), null, null)),
+      RetryStrategies.retryOnConflict(20, 100)
+    );
+  }
+
+  /**
+   * Adds datasets and types to the given {@link DatasetFramework} used by {@link OwnerStore}
+   *
+   * @param framework framework to add types and datasets to
+   */
+  public static void setupDatasets(DatasetFramework framework) throws IOException, DatasetManagementException {
+    framework.addInstance(Table.class.getName(), DATASET_ID, DATASET_PROPERTIES);
+  }
+
+  /**
+   * Store the owner's principal for the given {@link EntityId}
+   *
+   * @param entityId The {@link EntityId} whose owner principal needs to be stored
+   * @param principal the principal of the {@link EntityId} owner
+   * @throws IOException if failed to get the store
+   */
+  public void add(final EntityId entityId, final String principal) throws IOException, AlreadyExistsException {
```

Comment:
@chtyim :Yes, that seems like a good idea. Specifically, this is owner's principal and we also want to add authorization on it to determine who can impersonate whom. It might be better if we make user(owner) principal a cdap entity. What do you think?